lib/gist.rb

```diff
@@ -59,13 +59,17 @@ def self.write(token)
         f.write token
       end
     end
+
+    def self.read_token_or_fetch_from_env
+      ENV['GITHUB_AUTH_TOKEN'] || self.read rescue nil
```

Comment:
Variable name should be namespaced somehow to prevent conflicts. For instance, `GIST_GITHUB_API_TOKEN` or something.